packages/presentation/lib/home_page.dart

```diff
@@ -12,82 +13,82 @@ class MyHomePage extends StatefulWidget {
 }
 
 class _MyHomePageState extends State<MyHomePage> {
-  late TextEditingController _palindrome;
-  bool _result = true;
-  CheckingIfPalindrome checking = CheckingIfPalindrome();
+  HomeBloc bloc = HomeBloc(PalindromeUseCase());
 
   @override
   void initState() {
     super.initState();
-    _palindrome = TextEditingController();
-
+    bloc.initState();
   }
 
-  @override
-  void dispose() {
-    super.dispose();
-    _palindrome.dispose();
-
+  void _checkString() {
+    bloc.isPalindrome();
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-
         title: Text(widget.title),
       ),
-      body: Center(
+      body: StreamBuilder(
+          stream: bloc.dataStream,
+          initialData: BlocData.init(),
+          builder: (BuildContext context, AsyncSnapshot<dynamic> snapshot) {
+            final blocData = snapshot.data;
+            if (blocData is BlocData) {
+              final screenData = blocData.data;
+              if (blocData.isLoading) {
+                return const Center(child: CircularProgressIndicator());
+              } else if (screenData is HomeData) {
+                return _buildResult(screenData);
+              } else {
+                return Container(
+                  color: Colors.red,
+                );
+              }
+            } else {
+              return Container(
+                color: Colors.green,
+              );
+            }
+          }),
```

Comment:
in the bigger project stream builder should be encapsulated in a separate widget as I showed u. but now it's ok